src/na/device/Device.cpp

```diff
@@ -0,0 +1,499 @@
+/*
+ * Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+ * Copyright (c) 2025 Munich Quantum Software Company GmbH
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+/** @file
+ * @brief The MQT QDMI device implementation for neutral atom devices.
+ */
+
+#include "na/device/Device.hpp"
+
+#include "Initializer.hpp"
```

Comment:
This header confuses me. Can we give it a slightly more appropriate name? Could we perhaps namespace it properly?